### PR TITLE
Shopper Insights Final Changes

### PR DESCRIPTION
**Pull request by jaxdesmarais**
### Summary of changes

- Update `release.yml` JSON output file name
- Update CHANGELOG
- Update spacing in `PayPalWebCheckoutViewController`
- Remove old `BraintreeDemoPreferredPaymentMethodViewController`
- Cleanup syntax in `ShopperInsightsViewController`
- Sort files by name
- Update error enum docstring
- Update `paypal` to `payPal` in `BTEligiblePaymentMethods`
- Update typo in `BTEligiblePaymentMethods`
- Move tests to `UnitTest` scheme

### Checklist

- ~[ ] Added a changelog entry~

### Authors

- @jaxdesmarais

### CODE REVIEW

File: Braintree.xcodeproj/xcshareddata/xcschemes/UnitTests.xcscheme [hunk 1]
```diff
@@ -173,6 +173,16 @@
...
+            skipped = "NO">
+            <BuildableReference
+               BuildableIdentifier = "primary"
+               BlueprintIdentifier = "8046983D2B27C5530090878E"
+               BuildableName = "BraintreeShopperInsightsTests.xctest"
+               BlueprintName = "BraintreeShopperInsightsTests"
+               ReferencedContainer = "container:Braintree.xcodeproj">
+            </BuildableReference>
+         </TestableReference>
       </Testables>
```
Comment (KunJeongPark) [inline Braintree.xcodeproj/xcshareddata/xcschemes/UnitTests.xcscheme:186]: what does this do, Jax?

**Review by sshropshire — APPROVED** [review]

File: UnitTests/BraintreeCoreTests/Analytics/BTAnalyticsService_Tests.swift [hunk 2]
```diff
@@ -19,7 +19,7 @@ final class BTAnalyticsService_Tests: XCTestCase {
         
         await analyticsService.performEventRequest("any.analytics.event")
         
-        XCTAssertEqual(analyticsService.http?.customBaseURL?.absoluteString, "https://api-m.paypal.com")
```
Comment (KunJeongPark) [inline UnitTests/BraintreeCoreTests/Analytics/BTAnalyticsService_Tests.swift:22]: When was this changed?

Comment (jaxdesmarais) [reply UnitTests/BraintreeCoreTests/Analytics/BTAnalyticsService_Tests.swift:22]: PR: https://github.com/braintree/braintree_ios/pull/1175

File: Braintree.xcodeproj/xcshareddata/xcschemes/UnitTests.xcscheme (hunk 1 above)
Comment (jaxdesmarais) [reply Braintree.xcodeproj/xcshareddata/xcschemes/UnitTests.xcscheme:186]: The `BraintreeShopperInsightsTests` were listed under this section:
Screenshot 2024-06-06 at 2 25 52 PM

This change adds them under this section with the rest of our Unit Tests:
Screenshot 2024-06-06 at 2 26 51 PM

**Pull request merged**